riscv-executor/src/lib.rs

```diff
@@ -292,13 +334,14 @@ mod builder {
         /// May fail if max_rows_len is too small or if the main machine is
         /// empty. In this case, the final (empty) execution trace is returned
         /// in Err.
+        #[allow(clippy::type_complexity)]
```

Comment:
Think it wouldn't hurt to introduce a struct.

Reply:
idk, maybe a principle thing but I don't really see the point. I think a struct just adds overhead since the fields are rather unrelated. In the end we'd end up with a `struct ArgsToTraceBuilderNew` that exists solely for that reason.

Reply:
I meant introducing a type for `(ExecutionTrace<F>, MemoryState, RegisterMemoryState<F>)`. That does appear in other places and I think it would let you get rid of the annotation.